Refreshed documentation so it reflects how users should interact with… (#266)
casper-network/casper-sidecar: USAGE.md

The Sidecar event stream is a passthrough for all the events emitted by the node(s) to which the Sidecar connects. This stream also includes one endpoint for Sidecar-generated events that can be useful, although the node did not emit them.

Events are emitted on two endpoints: 

> * All events from that come from nodes are re-emitted under `http://<HOST>:<SIDECAR_SSE_PORT>/events`
> * All sidecar generated events (which are informing about the internal state of sidecar) are emitted under `http://<HOST>:<SIDECAR_SSE_PORT>/events/sidecar`

Reviewer: ```suggestion
* All events that come from a node are re-emitted under `http://<HOST>:<SIDECAR_SSE_PORT>/events`.
* All Sidecar-generated events reporting the Sidecar's internal state are emitted under `http://<HOST>:<SIDECAR_SSE_PORT>/events/sidecar`.
```

Author: Ok

It is possible to monitor the Sidecar event stream using *cURL*, depending on how the HOST and PORT are configured.

```json
curl -s http://<HOST:PORT>/events/<TYPE>
curl -s http://<HOST:PORT>/events
```

- `HOST` - The IP address where the Sidecar is running
- `PORT` - The port number where the Sidecar emits events
- `TYPE` - The type of event emitted

Given this [example configuration](./resources/example_configs/EXAMPLE_NODE_CONFIG.toml), here are the commands for each endpoint:

- **Deploy events:** 

    ```json
    curl -sN http://127.0.0.1:19999/events/deploys
    ```

- **Finality Signature events:** 

    ```json
    curl -sN http://127.0.0.1:19999/events/sigs
    ```

- **Main events:** 

    ```json
    curl -sN http://127.0.0.1:19999/events/main
    ```

- **Sidecar-generated events:** 

    ```json
    curl -sN http://127.0.0.1:19999/events/sidecar
    ```

> 

Reviewer: I'd like to add 2 examples:

Given this [example configuration](./resources/example_configs/EXAMPLE_NODE_CONFIG.toml), here are the commands for each endpoint:

- **Node events:** 

    ```json
    curl -sN http://127.0.0.1:19999/events
    ```

- **Sidecar-generated events:** 

    ```json
    curl -sN http://127.0.0.1:19999/events/sidecar
    ```

### The Node Shutdown Event

When the node sends a Shutdown event and disconnects from the Sidecar, the Sidecar will report it as part of the event stream and on the `/events` endpoint. The Sidecar will continue to operate and attempt to reconnect to the node according to the `max_attempts` and `delay_between_retries_in_seconds` settings specified in its configuration.

> Sidecar does not expose Shutdown events via its REST API. 

Reviewer: We need to use "the" whenever we refer to the Sidecar.

```suggestion
The Sidecar does not expose Shutdown events via its REST API. 
```
